qcodes/utils/deprecate.py

```diff
@@ -69,7 +69,13 @@ def actual_decorator(obj: Any) -> Any:
             for m_name in dir(obj):
                 m = getattr(obj, m_name)
                 if isinstance(m, (types.FunctionType, types.MethodType)):
-                     # pylint: disable=no-value-for-parameter
+                    # skip static methods, since they are not wrapped corectly
```

Comment:
> When deprecating a classmethod, a staticmethod, or a property, the    ``@deprecated`` decorator should go *under* the ``@classmethod``, etc.    decorator (i.e., `deprecated` should directly decorate the underlying    callable).

taken from here https://matplotlib.org/3.1.1/_modules/matplotlib/cbook/deprecation.html

Reply:
Yes, that is true, but `wrapt` should be addressing exactly that. And when decorating a class, there is no way (or at least not any obvious one) to programmatically replace the already defined functions, including their decorator with a version with a changed order of decorators.